// @kiwicom/orbit-react-native/src/Text/Text.styles.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = generateTextStyles;

function generateTextStyles(theme) {
  return {
    'base_text-normal-normal': {
      fontFamily: theme.fontFamily[theme.orbit.fontWeightNormal],
      color: theme.orbit.colorTextPrimary
    },
    'base_text-bold-normal': {
      fontFamily: theme.fontFamily[theme.orbit.fontWeightBold],
      color: theme.orbit.colorTextPrimary
    },
    'base_text-normal-italic': {
      fontFamily: theme.fontFamily["".concat(theme.orbit.fontWeightNormal, "-italic")],
      color: theme.orbit.colorTextPrimary
    },
    'base_text-bold-italic': {
      fontFamily: theme.fontFamily["".concat(theme.orbit.fontWeightBold, "-italic")],
      color: theme.orbit.colorTextPrimary
    },
    fontsizetext_small: {
      fontSize: parseFloat(theme.orbit.fontSizeTextSmall),
      lineHeight: parseFloat(theme.orbit.fontSizeTextSmall) * parseFloat(theme.orbit.lineHeightText)
    },
    fontsizetext_normal: {
      fontSize: parseFloat(theme.orbit.fontSizeTextNormal),
      lineHeight: parseFloat(theme.orbit.fontSizeTextNormal) * parseFloat(theme.orbit.lineHeightText)
    },
    fontsizetext_large: {
      fontSize: parseFloat(theme.orbit.fontSizeTextLarge),
      lineHeight: parseFloat(theme.orbit.fontSizeTextLarge) * parseFloat(theme.orbit.lineHeightText)
    }
  };
}
