// @kiwicom/orbit-react-native/src/Heading/Heading.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Heading = Heading;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _react = _interopRequireDefault(require("react"));

var _reactNative = require("react-native");

var _styles = require("../styles");

var _testUtils = require("../common/testUtils");

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(source, true).forEach(function (key) { (0, _defineProperty2.default)(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(source).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

// @TODO dataA11ySection once we have <SkipNavigation>
function Heading(_ref) {
  var _ref$type = _ref.type,
      type = _ref$type === void 0 ? 'title1' : _ref$type,
      _ref$inverted = _ref.inverted,
      inverted = _ref$inverted === void 0 ? false : _ref$inverted,
      _ref$spaceAfter = _ref.spaceAfter,
      spaceAfter = _ref$spaceAfter === void 0 ? 'none' : _ref$spaceAfter,
      _ref$as = _ref.as,
      as = _ref$as === void 0 ? 'h1' : _ref$as,
      id = _ref.id,
      dataTest = _ref.dataTest,
      children = _ref.children;
  var styles = (0, _styles.useStyles)({
    base: 'heading',
    headingType: type,
    inverted: inverted,
    spaceAfter: spaceAfter
  });

  var nextProps = _objectSpread({
    style: styles,
    nativeID: id,
    accessibilityRole: _reactNative.Platform.OS === 'web' ? 'heading' : 'header',
    'aria-level': '1'
  }, (0, _testUtils.testAttribute)(dataTest));

  if (as !== 'div') {
    switch (as) {
      case 'h5':
        nextProps['aria-level'] = '5';
        break;

      case 'h4':
        nextProps['aria-level'] = '4';
        break;

      case 'h3':
        nextProps['aria-level'] = '3';
        break;

      case 'h2':
        nextProps['aria-level'] = '2';
        break;

      case 'h1':
        nextProps['aria-level'] = '1';
        break;

      default:
        as;
    }
  } else {
    nextProps.accessibilityRole = '';
    nextProps['aria-level'] = '';
  }

  return _react.default.createElement(_reactNative.Text, nextProps, children);
} // Display name for Storybook


Heading.displayName = 'Heading';
