// @kiwicom/orbit-react-native/src/Illustration/Illustration.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Illustration = Illustration;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _react = _interopRequireDefault(require("react"));

var _styles = require("../styles");

var _Image = require("../Image");

var _Illustration = require("./Illustration.consts");

var _testUtils = require("../common/testUtils");

var _getIllustration2 = _interopRequireDefault(require("./utils/getIllustration"));

var _ThemeContext = _interopRequireDefault(require("../ThemeProvider/ThemeContext"));

function getIllustrationURL(name, height, heightMultiplier) {
  return "".concat(_Illustration.baseURL, "/illustrations/0x").concat(height * heightMultiplier, "/").concat(name, ".png");
}

function getHeightToken(_ref) {
  var theme = _ref.theme,
      size = _ref.size;
  var tokens = {
    small: theme.orbit.heightIllustrationSmall,
    medium: theme.orbit.heightIllustrationMedium
  };
  return size && tokens[size];
}

function Illustration(_ref2) {
  var _getIllustration;

  var dataTest = _ref2.dataTest,
      _ref2$size = _ref2.size,
      size = _ref2$size === void 0 ? 'medium' : _ref2$size,
      _ref2$spaceAfter = _ref2.spaceAfter,
      spaceAfter = _ref2$spaceAfter === void 0 ? 'none' : _ref2$spaceAfter,
      name = _ref2.name;
  var styles = (0, _styles.useStyles)({
    spaceAfter: spaceAfter,
    size: size,
    base: 'illustration'
  });

  var _React$useContext = _react.default.useContext(_ThemeContext.default),
      _React$useContext2 = (0, _slicedToArray2.default)(_React$useContext, 1),
      theme = _React$useContext2[0];

  var height = parseInt(getHeightToken({
    theme: theme,
    size: size
  }), 10);
  return _react.default.createElement(_Image.Image, (0, _extends2.default)({
    nativeSource: (_getIllustration = (0, _getIllustration2.default)(name)) !== null && _getIllustration !== void 0 ? _getIllustration : undefined,
    webSource: {
      '1': getIllustrationURL(name, height, 1),
      '2': getIllustrationURL(name, height, 2)
    },
    style: styles
  }, (0, _testUtils.testAttribute)(dataTest)));
} // Display name for Storybook


Illustration.displayName = 'Illustration';
