// @kiwicom/orbit-react-native/src/common/ForceRTL.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ForceRTL = ForceRTL;

var _react = _interopRequireDefault(require("react"));

var _reactNative = require("react-native");

var _orbitReactNative = require("@kiwicom/orbit-react-native");

var _orbitDesignTokens = require("@kiwicom/orbit-design-tokens");

/* eslint-disable no-undef */
var isWeb = typeof window !== 'undefined' && window.document && window.document.createElement;

function ForceRTL(_ref) {
  var children = _ref.children;

  _react.default.useEffect(function () {
    if (isWeb) {
      window.document.documentElement.dir = 'rtl';
    } else {
      _reactNative.I18nManager.forceRTL(true);
    }

    return function cleanup() {
      if (isWeb) {
        window.document.documentElement.dir = 'ltr';
      } else {
        _reactNative.I18nManager.forceRTL(false);
      }
    };
  });

  return _react.default.createElement(_orbitReactNative.ThemeProvider, {
    theme: {
      orbit: _orbitDesignTokens.defaultTokens,
      rtl: true,
      fontFamily: {
        '400': 'CircularPro-Book',
        '400-italic': 'CircularPro-BookItalic',
        '500': 'CircularPro-Medium',
        '700': 'CircularPro-Bold',
        // Map bold & italic text to be just bold for readability
        '500-italic': 'CircularPro-Medium',
        '700-italic': 'CircularPro-Bold'
      }
    }
  }, _react.default.createElement(_reactNative.View, {
    style: styles.container
  }, children));
}

var styles = {
  container: {
    width: '100%'
  }
};
