// @kiwicom/orbit-react-native/src/Modal/Modal.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Modal = Modal;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _react = _interopRequireDefault(require("react"));

var _reactNative = require("react-native");

var _ThemeContext = require("../ThemeProvider/ThemeContext");

var _utils = require("../common/utils");

var _testUtils = require("../common/testUtils");

function Modal(_ref) {
  var _ref$size = _ref.size,
      size = _ref$size === void 0 ? 'normal' : _ref$size,
      _ref$isMobileFullPage = _ref.isMobileFullPage,
      isMobileFullPage = _ref$isMobileFullPage === void 0 ? false : _ref$isMobileFullPage,
      dataTest = _ref.dataTest,
      children = _ref.children;

  var _React$useContext = _react.default.useContext(_ThemeContext.ThemeContext),
      _React$useContext2 = (0, _slicedToArray2.default)(_React$useContext, 1),
      theme = _React$useContext2[0];

  var borderRadiusModal = isMobileFullPage ? 9 : 0; // @TODO theme.orbit.borderRadiusModal
  // @TODO Instead of nested <RNModal>s, a single <RNModal> with some extra animations on a <RNView> inside should do much better.

  return _react.default.createElement(_reactNative.Modal, {
    animationType: "fade",
    transparent: true,
    visible: true
  }, _react.default.createElement(_reactNative.View, {
    style: {
      position: 'absolute',
      width: '100%',
      height: '100%',
      backgroundColor: 'rgba(0, 0, 0, 0.5)'
    }
  }, _react.default.createElement(_reactNative.Modal, {
    animationType: "slide",
    transparent: true,
    visible: true
  }, _react.default.createElement(_reactNative.SafeAreaView, {
    style: {
      flex: 1
    }
  }, _react.default.createElement(_reactNative.ScrollView, (0, _extends2.default)({
    style: {
      flex: 1,
      flexDirection: 'column',
      backgroundColor: theme.orbit.backgroundModal,
      borderTopLeftRadius: borderRadiusModal,
      borderTopRightRadius: borderRadiusModal
    },
    contentContainerStyle: {
      justifyContent: 'center',
      alignItems: 'center'
    },
    nestedScrollEnabled: true
  }, (0, _testUtils.testAttribute)(dataTest)), _react.default.createElement(_reactNative.View, {
    style: {
      flex: 1,
      width: parseFloat(theme.orbit["widthModal".concat((0, _utils.capitalize)(size))]),
      maxWidth: '100%'
    }
  }, children))))));
} // Display name for Storybook


Modal.displayName = 'Modal';
