// @kiwicom/orbit-react-native/src/Image/Image.web.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Image = Image;

var _typeof2 = _interopRequireDefault(require("@babel/runtime/helpers/typeof"));

var React = _interopRequireWildcard(require("react"));

var _reactNativeWeb = require("react-native-web");

function Image(_ref) {
  var style = _ref.style,
      source = _ref.source,
      webSource = _ref.webSource,
      alt = _ref.alt,
      title = _ref.title,
      dataTest = _ref.dataTest;
  var currentSource = webSource ? webSource : source;
  var uris = currentSource != null && (0, _typeof2.default)(currentSource) === 'object' ? Object.keys(currentSource).map(function (key) {
    return currentSource[key] ? "".concat(currentSource[key], " ").concat(key, "x") : null;
  }).join(',') : undefined;

  var ImageElement = function ImageElement(props) {
    return (0, _reactNativeWeb.createElement)('img', props);
  };

  return currentSource !== 'number' && React.createElement(ImageElement, {
    alt: alt,
    title: title,
    "data-test": dataTest,
    style: style,
    src: currentSource === 'object' ? currentSource[1] : currentSource,
    srcSet: uris
  });
}
